__tests__/gatewayProvider.test.ts

```diff
@@ -0,0 +1,211 @@
+        );
+
+        expect(receipt).toHaveProperty('transaction_hash');
+        expect(receipt).toHaveProperty('status');
+        expect(receipt).toHaveProperty('status_data');
+        expect(receipt).toHaveProperty('messages_sent');
+        expect(receipt).toHaveProperty('l1_origin_message');
+        expect(receipt).toHaveProperty('events');
+      });
+    });
+
+    describe('Contract methods', () => {
+      let deployResponse: DeployContractResponse;
+      let declareResponse: DeclareContractResponse;
+
+      beforeAll(async () => {
+        deployResponse = await provider.deployContract({ contract: compiledErc20 });
+        declareResponse = await provider.declareContract({ contract: compiledErc20 });
+        await Promise.all([
+          provider.waitForTransaction(deployResponse.transaction_hash),
+          provider.waitForTransaction(declareResponse.transaction_hash),
+        ]);
+      });
+
+      describe('deployContract', () => {
+        test('response', () => {
+          expect(deployResponse.contract_address).toBeTruthy();
+          expect(deployResponse.transaction_hash).toBeTruthy();
+        });
+      });
+
+      describe('declareContract', () => {
+        test('response', async () => {
+          expect(declareResponse.class_hash).toBeTruthy();
+          expect(declareResponse.transaction_hash).toBeTruthy();
+        });
+      });
+
+      describe('getClassAt', () => {
+        test('response', async () => {
+          // Hardcoded contract address as RPC node is throwing "Contract not found" error
+          const classResponse = await provider.getClassAt(
+            '0x0377cd03f6bed2ed201e9783f50b1b823d2ff65032da69f468bba5c6634dfbfa'
+          );
+          expect(classResponse).toHaveProperty('program');
+          expect(classResponse).toHaveProperty('entry_points_by_type');
+        });
+      });
+
+      describe('callContract', () => {
+        test('result', () => {
+          return expect(
+            provider
+              .callContract({
+                contractAddress: deployResponse.contract_address,
+                entrypoint: 'balance_of',
+                calldata: compileCalldata({
+                  user: '0x9ff64f4ab0e1fe88df4465ade98d1ea99d5732761c39279b8e1374fa943e9b',
+                }),
+              })
+              .then((res) => {
+                expect(Array.isArray(res.result)).toBe(true);
+              })
+          ).resolves.not.toThrow();
+        });
+      });
+    });
+  });
+
+  describe('Gateway specifc methods', () => {
+    const exampleTransactionHash =
+      '0x37013e1cb9c133e6fe51b4b371b76b317a480f56d80576730754c1662582348';
+
+    test('getContractAddresses()', async () => {
+      const { GpsStatementVerifier, Starknet } = await provider.getContractAddresses();
+      expect(typeof GpsStatementVerifier).toBe('string');
+      expect(typeof Starknet).toBe('string');
+    });
+
+    test('getTransactionStatus()', async () => {
```

Comment:
this is not supported by RPC?

Reply:
It's not part of the [spec](https://playground.open-rpc.org/?uiSchema%5BappBar%5D%5Bui:splitView%5D=false&schemaUrl=https://raw.githubusercontent.com/starkware-libs/starknet-specs/master/api/starknet_api_openrpc.json&uiSchema%5BappBar%5D%5Bui:input%5D=false&uiSchema%5BappBar%5D%5Bui:darkMode%5D=true&uiSchema%5BappBar%5D%5Bui:examplesDropdown%5D=false). Status is part of the transaction receipt response